app/weather_forecast.py

```diff
@@ -0,0 +1,265 @@
+from datetime import datetime, timedelta
+from dotenv import load_dotenv
+from os import getenv
+import requests
+
+
+""" This feature requires an API KEY - get yours free @ visual-crossing-weather.p.rapidapi.com """
+
+SUCCESS_STATUS = 0
+ERROR_STATUS = -1
+HISTORY_TYPE = "history"
+HISTORICAL_AVERAGE_TYPE = "historical-average"
+FORECAST_TYPE = "forecast"
+HISTORY_URL = "https://visual-crossing-weather.p.rapidapi.com/history"
+FORECAST_URL = "https://visual-crossing-weather.p.rapidapi.com/forecast"
+HEADERS = {
+    'x-rapidapi-host': "visual-crossing-weather.p.rapidapi.com"
+    }
+BASE_QUERY_STRING = {"aggregateHours": "24", "unitGroup": "metric", "dayStartTime": "00:00:01",
+                     "contentType": "json", "dayEndTime": "23:59:59", "shortColumnNames": "True"}
+HISTORICAL_AVERAGE_NUM_OF_YEARS = 3
+OUTPUT = {"Status": SUCCESS_STATUS, "ErrorDescription": None, "MinTempCel": None, "MaxTempCel": None,
+          "MinTempFar": None, "MaxTempFar": None, "Conditions": None, "ForecastType": None}
+
+
+def validate_date_input(day, month, year):
+    """ date validation.
+    Args:
+        day (int / str) - day part of date.
+        month (int / str) - month part of date.
+        year (int / str) - year part of date.
+    Returns:
+        (bool) - validate ended in success or not.
+        day (int) - day part of date.
+        month (int) - month part of date.
+        year (int) - year part of date.
+    """
+    try:
+        day = int(day)
+        month = int(month)
+        year = int(year)
+    except ValueError:
+        return False, day, month, year
+    if 1975 <= year <= 2050:
+        try:
+            datetime(year=year, month=month, day=day)
+        except ValueError:
+            return False, day, month, year
+    else:
+        return False, day, month, year
+    return True, day, month, year
+
+
+def get_data_from_api(url, input_query_string):
+    """ get the relevant weather data by calling the "Visual Crossing Weather" API.
+    Args:
+        url (str) - API url.
+        input_query_string (dict) - input for the API.
+    Returns:
+        success_in_get_weather_data (bool) - did the API call ended in success or failure (location not found etc).
+        response_json (json dict) - relevant part (data / error) of the JSON returned by the API.
+    """
+    load_dotenv()
+    HEADERS['x-rapidapi-key'] = getenv('WEATHER_API_KEY')
+    success_in_get_weather_data = True
+    response = requests.request("GET", url, headers=HEADERS, params=input_query_string)
+    try:
+        response_json = response.json()["locations"]
+    except KeyError:
+        success_in_get_weather_data = False
+        response_json = response.json()
+    return success_in_get_weather_data, response_json
```

Comment:
Check whether the request retuned a valid 200 response. There is also no need for a custom success boolean.
```
    response = requests.request("GET", url, headers=HEADERS, params=input_query_string)
    if response:
        try:
            return response.json()["locations"]
        except KeyError:
            return None, response.json()["message"]
    else:
        return None, "error message"   # TODO: valid error message
```